File: TechnicalTest/Assignment4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TechnicalTest
{
    public class Assignment4
    {
        public static char[] Flatten(char[] array)
        {
            StringBuilder builder = new StringBuilder();

            foreach (char item in array)
            {
                if (Char.IsDigit(item))
                {
                    builder.Append(item);
                }
            }

            return builder.ToString().ToCharArray();
        }

        public void Flatten_ValidArray()
        {
            var array = new char[] { '(', '1', '2', '3', '3', ')' };

            Flatten(array);
        }
    }

}

File: TechnicalTestVB/Assignment4.vb
Imports System.Text

Public Class Assignment4

    Public Shared Function Flatten(array As Char()) As Char()
        Dim builder As New StringBuilder()

        For Each item As Char In array
            If [Char].IsDigit(item) Then
                builder.Append(item)
            End If
        Next

        Return builder.ToString().ToCharArray()
    End Function
End Class
